core/swift3Action/spm-build/_Whisk.swift

```diff
@@ -24,27 +24,40 @@ class Whisk {
         let strBlocking = blocking ? "true" : "false"
         let path = "/api/v1/namespaces/\(parsedAction.namespace)/actions/\(parsedAction.name)?blocking=\(strBlocking)"
 
-        return postSyncronish(uriPath: path, params: params)
+        return postOrPutSyncronish(uriPath: path, params: params, method: "post")
     }
 
     class func trigger(eventNamed event : String, withParameters params : [String:Any]) -> [String:Any] {
         let parsedEvent = parseQualifiedName(name: event)
         let path = "/api/v1/namespaces/\(parsedEvent.namespace)/triggers/\(parsedEvent.name)?blocking=true"
 
-        return postSyncronish(uriPath: path, params: params)
+        return postOrPutSyncronish(uriPath: path, params: params, method: "post")
+    }
+
+    class func createTrigger(triggerNamed trigger: String, withParameters params : [String:Any]) -> [String:Any] {
+        let parsedTrigger = parseQualifiedName(name: trigger)
+        let path = "/api/v1/namespaces/\(parsedTrigger.namespace)/triggers/\(parsedTrigger.name)"
+        return postOrPutSyncronish(uriPath: path, params: params, method: "put")
+    }
+
+    class func createRule(named ruleName: String, withTrigger trigger: String, andAction actionName: String) -> [String:Any] {
+        let parsedRule = parseQualifiedName(name: ruleName)
+        let path = "/api/v1/namespaces/\(parsedRule.namespace)/rules/\(parsedRule.name)"
+        let params = ["trigger":trigger, "action":actionName]
+        return postOrPutSyncronish(uriPath: path, params: params, method: "put")
     }
 
     // handle the GCD dance to make the post async, but then obtain/return
     // the result from this function sync
-    private class func postSyncronish(uriPath path: String, params : [String:Any]) -> [String:Any] {
+    private class func postOrPutSyncronish(uriPath path: String, params : [String:Any], method: String) -> [String:Any] {
```

Comment:
Yep, I'll change the api name tomorrow if that's ok?

Probably something like `sendWhiskRequestSyncronish`